src/components/Swap/Swap.tsx

```diff
@@ -628,21 +612,18 @@ const Swap: React.FC<{
               <Box />
             )}
             <Button
-              id='remove-recipient-button'
               onClick={() => onChangeRecipient(recipient !== null ? null : '')}
             >
               {recipient !== null ? '- Remove send' : '+ Add a send (optional)'}
             </Button>
           </Box>
           {recipient !== null && (
-            <Box className='content'>
-              <Typography>Recipient</Typography>
-              <input
-                value={recipient}
-                placeholder='Wallet Address or ENS name'
-                onChange={(evt) => onChangeRecipient(evt.target.value)}
-              />
-            </Box>
+            <AddressInput
```

Comment:
We should check if the address is ENS, or implement a checksum to validate the address.  

https://ethereum.stackexchange.com/questions/1374/how-can-i-check-if-an-ethereum-address-is-valid

Reply:
that logic is in solidity but should be applicable to javascript

Reply:
We are checking if the address is checksummed in `useENS` hook.